### ISSUE TITLE
constructor pattern allowed in meta.function-call

### ISSUE THREAD

**Issue by matter123** [jeff-hykin/cpp-textmate-grammar#261]
Example:
```c++
constexpr A *__A() {
	return A() ? A::get<0>(A)._l.A
                 : A::A(A::get<0>(A)._s.A[0]);
}
constexpr void func() {
}
```

Image:
![Screenshot from 2019-06-26 11-21-41](https://user-images.githubusercontent.com/714007/60204930-d4dc2880-9804-11e9-9fd1-239d68fa77e8.png)

`A::A(/*snip*/)` is treated as a constructor definition. This causes the subsequent function definition to be treated as part of the parameters.

**Comment by jeff-hykin** [comment]
Good catch, I forgot to remove ```
:constructor_root,
:destructor_root,
:operator_overload
```
when they were added to the root context

**Comment by jeff-hykin** [comment]
fixed in `v1.12.3`
